### ISSUE TITLE
Backreference WC : How to Copy the first characters of my whole try?

### ISSUE THREAD

**Issue by lyncker** [gurnec/btcrecover#242]
Hi , I have the problem that my password may have the the same sign at the very beginning like in the end... So i just want to try the same sign , if one is tried like the following: 

!mytest tokens a long!
?but the length of the tokens may different?
$this is illegal! << allways the the same beginning and end character !

b/c the length of my tries is different in lengh , I can not use the backreference wildcard which allways copies a char of a given length... 

any ideas? 

Thanks a lot...

**Comment by pedrohrcunha** [comment]
That's very simple. Add the following tokens in their own lines:

+ ^! [or any symbol you want to test at the beginning]
+ !$ [or any symbol you want to test at the end]

If you want to test for "$" (it is a special character, it anchors the token in the last position), try this:

+ $$$

If it doesn't work, try:

+ ^4^$$

"4" assuming to be the last position of a 4 token password.

**Comment by lyncker** [comment]
thanks pedrohrcunha for your answer, but maybe we misunderstood: 

I need to test only the cases when the beginning char is the same linke the ending char ... In your explanation I cant see that your ending char is referencing to the beginning.. I know that ^! is the beginnng and !$ the end... but how can I minimise the tries so that only the same sign in the beginng like in the end is tested ...

**Comment by pedrohrcunha** [comment]
@lyncker 

Oh, I see, use the "+". The "+" obliges the symbols occurrences in the beginning and in the end of the password. 

So that would be:

[password guesses]
                   + ^!
                   + !$

Works for any symbol. Our miscomunic was github's fault.

**Comment by lyncker** [comment]
ah ok , maybe I was to short... 
I have different symbols like *!"&/( etc .. they may be there or not... the only thing i know is , if for example the questionmark ? is in the beginning, it must be at the end ... so my tokenfile looks like: 
```

+ mytesttoken%s
+ anothertesttoken
 ^. ^* ^? ^< ^> 
 .$ *$ ?$ <$ >$ 


the desired results are : 
mytesttoken anothertesttoken
!mytesttoken anothertesttoken!
?mytesttoken anothertesttoken?
*mytesttoken anothertesttoken*

but NOT : 
*mytesttoken anothertesttoken!

```


therefore I need a way to reference the first char so that the ending char must be the same ... 

thanks again for your time !

**Comment by pedrohrcunha** [comment]
You will have to set multiple tokens and try separatly, one by one...

**Comment by lyncker** [comment]
That is not possible , due to my tokens are not static : For example I have : 
```
+ mytest%[ABC]token
+ anothe%drtest
```
so these tokens are composed and iterated in many different versions by themselves How can I add these special symbols at the beginning and the end ?

**Comment by pedrohrcunha** [comment]
@lyncker I don't follow. 

You can add these symbols necessairly at the beginning and at the end by writing the two lines we discussed above. They *will* occur in every password attempt. You will have to save a .txt for every probable symbol of your password. It's not a lot, as it seems. So:

       + mytest%[ABC]token
       + anothe%drtest
       + ^!
       + !$

Another .txt...

[The same thing, with other symbols.] 

I'm pretty sure there's a better solution, but I don't have the mind to put into it right now. Because time is money, unfortunately.

**Comment by lyncker** [comment]
yes thats what Im doing... then I have to check everything again with the next txt file which uses another symbol...unfortunately each run takes soooo long ... ;( hoped there is at least a solution which does it for me so I can leave the process alone ...
thanks anyway !

**Comment by pedrohrcunha** [comment]
@lyncker There are always better solutions. Give time to your thinking develop them. But keep in mind the fact that your endeavor is realistic or not.

**Comment by btcexchanging** [comment]
if you haven't recovered your wallet, i can help you recover your wallet.
I have recovered most of the common wallets and recovery phrases for many users
you can reach me here at my telegram account:  @walletrecover
